cns/restserver/internalapi_test.go

```diff
@@ -48,6 +48,60 @@ func TestCreateOrUpdateNCWithLargerVersionComparedToNMAgent(t *testing.T) {
 	validateCreateOrUpdateNCInternal(t, 2, "1")
 }
 
+func TestCreateAndUpdateNCWithSecondaryIPNCVersion(t *testing.T) {
+	restartService()
+
+	setEnv(t)
+	setOrchestratorTypeInternal(cns.KubernetesCRD)
+	// NC version set as 0 which is the default initial value.
+	ncVersion := 0
+	secondaryIPConfigs := make(map[string]cns.SecondaryIPConfig)
+	receivedSecondaryIPConfigs := make(map[string]cns.SecondaryIPConfig)
+	ncID := "testNc1"
+
+	// Build secondaryIPConfig, it will have one item as {IPAddress:"10.0.0.16", NCVersion: 0}
+	ipAddress := "10.0.0.16"
+	secIPConfig := newSecondaryIPConfig(ipAddress, ncVersion)
+	ipId := uuid.New()
+	secondaryIPConfigs[ipId.String()] = secIPConfig
+	req := createNCReqInternal(t, secondaryIPConfigs, ncID, strconv.Itoa(ncVersion))
+	containerStatus := svc.state.ContainerStatus[req.NetworkContainerid]
+	// Validate secondary IPs' NC version has been updated by NC request
+	receivedSecondaryIPConfigs = containerStatus.CreateNetworkContainerRequest.SecondaryIPConfigs
```

Comment:
- no need to make a map in line 59, just use `:=` here
- the reason why a check for existence of entry in the map is better is that the test won't fail if there is no match. If you don't get the check for existence, you get the zero value of a struct, including all of it's members recursively, and ranging over a nil map is valid. In case the nc id was not found in the map, the checks below will not happen, and the test will continue as normal. Let me know if that makes sense.

Reply:
I think what you are trying to say is adding one more check in map length. I can do that though it's not the key point in this test.